### DISCUSSION TITLE
Regarding G-Helper update to v0.173 some Laptop Keyboard are missing

### DISCUSSION THREAD

**Discussion by 1-metre-Sunshine** [seerge/g-helper#2691] [Q&A] [answered]
The following figure shows G-Helper v0.170, Laptop Keyboard including Rain.
![v170](https://github.com/seerge/g-helper/assets/61453424/641db468-ec12-46c7-9055-6e03f7cfcff2)

The following figure shows G-Helper v0.173, where the Laptop Keyboard does not include Rain and some available options are missing.
![v173](https://github.com/seerge/g-helper/assets/61453424/b69d892d-2b0f-41ba-b658-139503eb3880)

**Comment by seerge** [comment]
@1-metre-Sunshine I have removed some modes not supported by hardware. You have 4-zone RGB model - did that "Rain" do anything on your device ?

**Reply by 1-metre-Sunshine** [reply]
What I'm wondering is, the Rain does light up, so why does it say that the hardware doesn't support it?

The key point is that I've been using the Rain Laptop Keyboard since I bought my computer and I'm accustomed to the light effect.
But I'd say it's just a habit, not a necessity.

**Reply by seerge** [reply]
@1-metre-Sunshine how does rain look on your keyboard ?

**Reply by seerge** [reply]
@1-metre-Sunshine ok, so you actually have per-key RGB.

Looks like your exact model G614JI exists in 2 editions (per-key RGB and 4-zone RGB), and there is no way to distinguish them.

Do Ambient / Heatmap modes work ok on your device ?

**Reply by 1-metre-Sunshine** [reply]
Ambient / Heatmap modes can only display a single color and cannot modify the color.

Oh, and I just noticed that Rain mode can be displayed but cannot change the color on my device, does this mean that the hardware doesn't support it?

Then I understand why you're considering removing them.

**Reply by seerge** [reply]
@1-metre-Sunshine ambient mode should reflect contents of your screen, does it work and do that in 0.173 ?

**Reply by seerge** [reply, marked as answer]
@1-metre-Sunshine try this build

Since your model exists in different options, you need to open ``Extra`` settings and check there `Per-key RGB Keyboard` checkbox.
After checking - just restart app.

[GHelper.zip](https://github.com/user-attachments/files/15587694/GHelper.zip)

**Reply by seerge** [reply]
@1-metre-Sunshine did the build i linked here help ?

**Reply by 1-metre-Sunshine** [reply]
Functionally it can be basically the same as version v0.170.

**Reply by seerge** [reply]
Great

**Reply by squee666** [reply]
> @1-metre-Sunshine did the build i linked here help ?

possible to piggy back here because my GX650 / PZ zeph dou doesnt have per key options in ghelper but its specifically a per key only model

**Reply by seerge** [reply]
@squee666 I have added this checkbox only for models that exist in both per-key and 4-zone RGB (it's literally only a few SKUs). There i simply can't separate them - and therefore added this option to the UI (before it existed in config)

And for the rest - app already knows what kind of backlight you have. Do all modes work correctly on your device (like Ambient)? Or some are missing / not working ?

**Reply by squee666** [reply]
> @squee666 I have added this checkbox only for models that exist in both per-key and 4-zone RGB (it's literally only a few SKUs). There i simply can't separate them - and therefore added this option to the UI (before it existed in config)
> 
> And for the rest - app already knows what kind of backlight you have. Do all modes work correctly on your device (like Ambient)? Or some are missing / not working ?

ambient isnt working althought i dont know what its meant to do lol
just incase I installed armoury crate again to find my missing modes

![image](https://github.com/seerge/g-helper/assets/628576/07743392-f674-430a-aeb7-8d9ce451c7e2)
![image](https://github.com/seerge/g-helper/assets/628576/0f0a872f-9268-4dd4-9fcb-9c7ab3c06ac8)

**Reply by seerge** [reply]
@squee666 can you try this build ? 

1. It should show more modes (check if they work)
2. Check if Ambient and Heatmap work here ? Ambient should set keyboard color into colors of your screen

[GHelper.zip](https://github.com/user-attachments/files/15840300/GHelper.zip)

**Reply by squee666** [reply]
> @squee666 can you try this build ?
> 
> 1. It should show more modes (check if they work)
> 2. Check if Ambient and Heatmap work here ? Ambient should set keyboard color into colors of your screen
> 
> [GHelper.zip](https://github.com/user-attachments/files/15840300/GHelper.zip)

yeah ambient works and all of the modes work except Comet ,Comet does the same Animation as Flash when selected

**Reply by seerge** [reply]
@squee666 great, I will include this in next release. I was not aware that Duo has per-key RGB (and therefore extended list of animations) :) Thanks for spotting.

As for Comet - on some models some animations intersect, that's fine.

**Reply by RamouzK** [reply]
> @1-metre-Sunshine try this build
> 
> Since your model exists in different options, you need to open `Extra` settings and check there `Per-key RGB Keyboard` checkbox. After checking - just restart app.
> 
> [GHelper.zip](https://github.com/user-attachments/files/15587694/GHelper.zip)

Sorry for adding a comment here a year later. I don't want to open an issue for this since it's a question I have about usage for this.
In my new G18 2025, I see Per-Key RGB Keyboard, checking it does nothing. How do I make use of this? I only use G-Helper. I didn't have this with my G18 2023.

**Reply by seerge** [reply]
@RamouzK 

Some laptop models with exactly same model name exist in 2 editions - with per-key-RGB and with 4-zone RGB. And since command to set colors on them is different - in that case I simply don't know in advance what command to use. 

For them by default, app would treat those models as 4-zone RGB, and just show this extra checkbox under `Extra` to enable (optional) per-zone-RGB command instead. So people can set it manually. 

This command is used only for Ambient and Heatmap modes (where G-Helper sets colors manually), all other modes are just automatic and baked into firmware and work same no matter of the keyboard type :)

**Reply by RamouzK** [reply]
@seerge - Thanks. I'm trying to understand the point of this. All the lighting modes work for me without that option ticked on my G18 2025. I don't need to tick the box for it, right?
Also, if I want to use per-key RGB, do I need to install Asus Aura or any other app? Or does G-Helper have that (that would be awesome)?

**Reply by seerge** [reply]
@RamouzK yes, if it works well - you don't need to tick it :) So likely you have 4-zone-RGB and your keyboard simply does not support setting individual colors on per-key basis.

**Reply by RamouzK** [reply]
@seerge - I took a look at their website. The information is all over the place. But, after confirming with other sources, their RTX 5060 and 5070 models have per-key, but their 5070 Ti (which is what I have) and the RTX 5080 have 4-zone RGB! But, we also get a different keyboard and style with clear translucent keys (QWER ASDF and space), while the others get an older style keyboard, from what I see. I do like the typing on this one more than my 2023 so this is a bonus.

Either way, that is so strange. Usually the higher end models have per-key. So, my model doesn't support per-key, weird.
